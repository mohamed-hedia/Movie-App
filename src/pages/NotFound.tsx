import Heading from "@/components/ui/Heading";

const NotFound = () => {
  return (
    <main className="w-full md:w-[calc(100%-8rem)] h-screen fixed md:right-0 px-5 md:px-0 flex justify-center items-center">
      <div className="flex flex-col items-center justify-center">
        <Heading as="h1" className="-mt-0 text-orange text-center leading-10">404<br/>Not Found</Heading>
      </div>
    </main>
  );
};
export default NotFound;
