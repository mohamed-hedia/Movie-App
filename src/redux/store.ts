import { configureStore } from "@reduxjs/toolkit";
import homeSlice from "./home/homeSlice";
import moviesSlice from "./movies/moviesSlice";
import tvSlice from "./tv/tvSlice";
import detailsSlice from "./details/detailsSlice";
import searchSlice from "./search/searchSlice";
import userSlice from "./user/userSlice";
import bookmarkSlice from "./bookmarked/bookmarkSlice";

export const store = configureStore({
  reducer: {
    user: userSlice,
    home: homeSlice,
    movies: moviesSlice,
    tv: tvSlice,
    details: detailsSlice,
    search: searchSlice,
    bookmark: bookmarkSlice,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
