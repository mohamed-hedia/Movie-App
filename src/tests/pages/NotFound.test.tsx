import NotFound from "@/pages/NotFound";
import { store } from "@/redux/store";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";


describe("NotFound Page", () => {
  test("renders NotFound Page should match the screenshot", () => {
    const notFound = render(
      <Provider store={store}>
        <Router>
          <NotFound />
        </Router>
      </Provider>
    );

    expect(notFound).toMatchSnapshot();
  });
});
