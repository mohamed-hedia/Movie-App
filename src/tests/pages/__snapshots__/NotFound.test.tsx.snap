// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound Page > renders NotFound Page should match the screenshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="w-full md:w-[calc(100%-8rem)] h-screen fixed md:right-0 px-5 md:px-0 flex justify-center items-center"
      >
        <div
          class="flex flex-col items-center justify-center"
        >
          <h1
            class="antialiased text-3xl font-outfitLight -mt-0 text-orange text-center leading-10"
          >
            404
            <br />
            Not Found
          </h1>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="w-full md:w-[calc(100%-8rem)] h-screen fixed md:right-0 px-5 md:px-0 flex justify-center items-center"
    >
      <div
        class="flex flex-col items-center justify-center"
      >
        <h1
          class="antialiased text-3xl font-outfitLight -mt-0 text-orange text-center leading-10"
        >
          404
          <br />
          Not Found
        </h1>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
